src/github.com/matrix-org/dendrite/clientapi/routing/whoami.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2017 Vector Creations Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package routing
+
+import (
+	"database/sql"
+	"net/http"
+
+	"github.com/matrix-org/dendrite/clientapi/auth/storage/devices"
+	"github.com/matrix-org/dendrite/clientapi/jsonerror"
+	"github.com/matrix-org/util"
+)
+
+type whoamiResponse struct {
+	UserID string `json:"user_id"`
+}
+
+// Whoami implements /account/whoami endpoint method GET
+// https://matrix.org/docs/spec/client_server/r0.3.0.html#get-matrix-client-r0-account-whoami
+func Whoami(
+	req *http.Request, deviceDB *devices.Database,
+) util.JSONResponse {
+
+	if req.Method == http.MethodGet {
+		token := req.URL.Query().Get("access_token")
```

Comment:
Instead of retrieving the `access_token` through the query, it will be sent through the `Authorization` header. See [this document](https://github.com/anoadragon453/dendrite/wiki/How-to-Test-with-Authentication-to-the-Client-Server-API) for clarification. We should pull it from there instead of the URL.

This code within Auth extracts the access token from the Authorization header: https://github.com/APwhitehat/dendrite/blob/b4ca664960a6bdf19dcb9ace7a4a4e74c46008dd/src/github.com/matrix-org/dendrite/clientapi/auth/auth.go#L98

Right now it's not public, but I'd suggest making it a public method and then using it here to get the token instead.